File: Trie/1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        int n = s.size();
        int i = 0;
        while (i < n) {
            int j = i + 1;
            while (j < n && s[j] == s[i] && j - i + 1 <= 2) {
                j += 1;
            }
            while (j < n && s[j] == s[i]) {
                s[j] = '#';
                j += 1;
            }
            i = j;
        }
        string ans;
        for(auto &ch : s) {
            if (ch != '#') ans += ch;
        }
        return ans;
    }
};

File: Trie/1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        n = len(s)
        i = 0
        ans = []
        while i < n:
            j = i + 1
            tmp = [s[i]]
            while len(tmp) < 2 and j < n and s[j] == s[j - 1]:
                tmp.append(s[j])
                j += 1  
            ans.extend(tmp)
            while j < n and s[j] == s[j - 1]:
                j += 1
            i = j
        return ''.join(ans)
            
